### ISSUE TITLE
[DOC]Documentation for subword_tokenize 

### ISSUE THREAD

**Issue by VibhuJawa** [rapidsai/cudf#5799]
## Report incorrect documentation

We need to improve our documentation for  [subword_tokenize](https://docs.rapids.ai/api/cudf/nightly/api.html#cudf.core.column.string.StringMethods.subword_tokenize), am starting this issue to document places where we need to improve it. 

**Describe the problems or issues found in the documentation**

We need to provide better documentation for the following parameters as these are new parameters that are not present in the defacto standard of [Hugging Face](https://huggingface.co/transformers/main_classes/tokenizer.html#pretrainedtokenizer) that we should follow.  

**Input Parameters:** 
All these are new parameters that are not present in HF 

- max_rows_tensor: The default for this also quite less(500) and is a new parameter that we introduce, we should look into better ways to document this 

Below is present in both HF and ours but we have a divergence 
- stride : (We might have a divergence when setting to 0 which is the HF default) , we either need to fix that or mention it in the documentation at least. 

Output Parameters: 
- metadataColumn: This also a new parameter that we have introduced here, we should look into providing an example as we have in the [blog](https://medium.com/rapids-ai/preprocess-your-training-data-at-lightspeed-with-our-gpu-based-tokenizer-for-bert-language-models-561cf9c46e15) 

#### Handling Special Characters:
We also seem to not be differently handling `special characters` in inputs currently but I am guessing that maybe stemming from on how we create hash_table.  (Related issue: https://github.com/rapidsai/cudf/issues/5765) . 

```python
cudf_tensor,_,_ = cudf.Series("a [SEP] b").str.subword_tokenize("vocab-hash.txt",do_lower=False)
cudf_tensor
```

```python
170,   164, 12342,  2101,   166,   171
# [SEP]-> Sep is broken down into 164, 12342,  2101, 166 ([', 'SE', '##P', '])  vs just `[SEP]` in HF. 
```

**Below have been fixed with [PR 5919](https://github.com/rapidsai/cudf/pull/5919) and [PR 6658](https://github.com/rapidsai/cudf/pull/6568)** 

-  hash_file: We need to provide documentation for how to create the hash file 

- max_num_strings:  We have undefined behavior if `input_strings > max_num_strings` and the default is really less (100).  This can lead to issues like in the one at [Related issue](https://github.com/rapidsai/cudf/issues/5796). 

- max_num_chars:  This also a new parameter that we introduce, so we should provide more info about this.

CC: @davidwendt / @raykallen / @BartleyR  / @randerzander

**Comment by VibhuJawa** [comment]
> * max_num_strings:  We have undefined behavior if `input_strings > max_num_strings` and the default is really less (100).  This can lead to issues like in the one at [Related issue](https://github.com/rapidsai/cudf/issues/5796).

I am not sure why we have this parameter, I think it might be worth throwing a logic error (Which I think should come at no extra cost) when the number of input strings is too large. Having this additional parameter with a low default value may trip other users too. The error we throw here is a `cudaErrorIllegalAddress` which does not inform the user about where it is stemming from.

**Comment by VibhuJawa** [comment]
Happy to take this on for `0.17`  release. Would love to clean this up. 

Related to this issue, I will need support on how to add the [hashingfile](https://github.com/rapidsai/clx/blob/267c6d30805c9dcbf80840f222bf31c5c4b7068a/python/clx/analytics/perfect_hash.py) to the `cudf` package to close part of this. 

Related  See open question at: https://github.com/rapidsai/cudf/issues/5868#issuecomment-670189653


CC: @kkraus14  .

**Comment by VibhuJawa** [comment]
Last of the issues will be addressed with #6608 .

**Comment by VibhuJawa** [comment]
Closed by https://github.com/rapidsai/cudf/pull/6608, https://github.com/rapidsai/cudf/pull/5919 and https://github.com/rapidsai/cudf/pull/6658
